src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -898,7 +918,12 @@ public struct OperationAnalysisContext
         /// <see cref="IOperation"/> that is the subject of the analysis.
         /// </summary>
         public IOperation Operation { get { return _operation; } }
-        
+
+        /// <summary>
+        /// <see cref="CodeAnalysis.SemanticModel"/> that can provide semantic information about the <see cref="IOperation.Syntax"/> associated with the <see cref="Operation"/>.
+        /// </summary>
+        public SemanticModel SemanticModel { get { return _semanticModel; } }
```

Comment:
Validating that a diagnostic reported by the analyzer has location within a compilation requires the analysis context to have the compilation being analyzed. All the context types that shipped prior to recently added Operation analysis contexts take either the semantic model/compilation parameter in the constructor and expose the same it publically on the context, but the Operation related contexts don't hold onto or expose either of these. This change makes the public API shape of Operation context types consistent with other context types by exposing the Semantic model for the tree corresponding to operation being analyzer and also passing down the owning compilation of the model to the code validating diagnostic location - I can rip out the public API changes by making these internal if you insist. @JohnHamby what is your take on this?